yjit/src/asm/mod.rs

```diff
@@ -274,14 +407,43 @@ impl CodeBlock {
         assert!(self.label_refs.is_empty());
     }
 
+    pub fn clear_labels(&mut self) {
+        self.label_addrs.clear();
+        self.label_names.clear();
+        self.label_refs.clear();
+    }
+
+    pub fn get_label_state(&self) -> LabelState {
+        LabelState {
+            label_addrs: self.label_addrs.clone(),
+            label_names: self.label_names.clone(),
+            label_refs: self.label_refs.clone(),
+        }
+    }
+
+    pub fn set_label_state(&mut self, state: LabelState) {
+        self.label_addrs = state.label_addrs;
+        self.label_names = state.label_names;
+        self.label_refs = state.label_refs;
+    }
```

Comment:
I don't like this kind of weird side-mutation that we do for the retry but I can't think of a better way of doing it 😅

Reply:
One idea is, now that we have `LabelState`, we could move some methods around and modify a temporary `LabelState` for each `Insn` instead of writing to `CodeBlock` and then "commit" the temporary `LabelState` to `CodeBlock`. But it's not that simpler than the current implementation anyway, so I feel like it's fair to leave it for another PR, merging this first to make other works easier.

Reply:
Yes no worry, this is good.